pandas/core/strings.py

```diff
@@ -823,6 +824,42 @@ def str_match(arr, pat, case=True, flags=0, na=np.nan):
     return _na_map(f, arr, na, dtype=dtype)
 
 
+def str_fullmatch(arr, pat, case=True, flags=0, na=np.nan):
```

Comment:
can you type

Reply:
Thanks for taking a look! I think your second comment might have gotten cut off. I've updated the "Testing for Strings that match or contain a pattern" section in `doc/source/user_guide/text.rst` to include information about `fullmatch` and how it differs from `match`. I've also added a `versionadded` annotation to the docstring for `str_fullmatch`. These changes are in commit 16bc011 . 

If there is anything else you would like changed, please let me know!

Reply:
The question is if you can add annotations, so should be something like:

```suggestion
def str_fullmatch(arr: ArrayLike, pat: Union[str, Pattern], case: bool=True, flags: int=0, na: Scalar=np.nan):
```

ArrayLike and Scalar would be importable from pandas._typing

Reply:
If you can add these then I think this should be good to go

Reply:
Sure, no problem at all. I've added those to `str_fullmatch`, and also updated `str_match` to match, in commit 3aadc20 .